src/state/transactions/reducer.ts

```diff
@@ -77,5 +79,12 @@ export default createReducer(initialState, (builder) =>
       } else if (transactions[chainId]?.[hash].type === 'limit-order-cancellation') {
         confirmOrderCancellation(chainId, receipt.from, hash, receipt.status !== 0)
       }
+    })
+    .addCase(resetUserState, (transactions, { payload: { chainId } }) => {
+      if (transactions[chainId]) {
+        transactions[chainId] = {}
+      }
+      const lsOrderKeys = getLocalStorageItemKeys(LS_ORDERS)
+      lsOrderKeys.forEach((lsOrderKey) => window?.localStorage?.removeItem(lsOrderKey))
```

Comment:
Hmm. it's not recommended that we should have side-effect operation in the reducer. We should keep it pure.

Add/Remove localStorage should be done in middleware or action instead.

Reply:
Done